File: canteen-app-models/src/main/java/com/example/canteen_app_models/models/FoodItem.java
package com.example.canteen_app_models.models;

import androidx.annotation.NonNull;

import com.example.canteen_app_models.helpers.FoodMenuDetails;

import java.io.Serializable;
import java.util.HashMap;
import java.util.Objects;

public class FoodItem implements Serializable {

    protected String itemName, itemPrice, itemCategory;
    protected long itemRating;

    public FoodItem(String itemName, String itemPrice, long itemRating, String itemCategory) {
        this.itemName = itemName;
        this.itemPrice = itemPrice;
        this.itemRating = itemRating;
        this.itemCategory = itemCategory;
    }

    public FoodItem() {
    }

    public static FoodItem fromMap(Object dsp, String category, String itemName) {
        HashMap<String, Object> map = (HashMap<String, Object>) dsp;
        if (map != null) {
            long rating = -1;
            if (map.containsKey(FoodMenuDetails.RATING)) {
                rating = (long) map.get(FoodMenuDetails.RATING);
            }
            return new FoodItem(itemName, String.valueOf(map.get(FoodMenuDetails.PRICE)),
                    rating,
                    category);
        } else {
            return null;
        }
    }

    public String getItemCategory() {
        return itemCategory;
    }

    public String getItemName() {
        return itemName;
    }

    public String getItemPrice() {
        return itemPrice;
    }

    public long getItemRating() {
        return itemRating;
    }

    @NonNull
    @Override
    public String toString() {
        String itemData = "";
        itemData = itemData.concat("\n\t\t Item Name: " + itemName
                + " Item Category: " + itemCategory + " Item Price: " + itemPrice
                + " Item Rating: " + itemRating);
        return "\n Item Data : " + itemData;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        FoodItem foodItem = (FoodItem) o;
        return itemRating == foodItem.itemRating &&
                itemName.equals(foodItem.itemName) &&
                Objects.equals(itemPrice, foodItem.itemPrice) &&
                Objects.equals(itemCategory, foodItem.itemCategory);
    }

    @Override
    public int hashCode() {
        return Objects.hash(itemName, itemPrice, itemCategory, itemRating);
    }
}

File: canteen-app-models/src/main/java/com/example/canteen_app_models/models/FoodItem.kt
package com.example.canteen_app_models.models

import com.example.canteen_app_models.helpers.FoodMenuDetails
import java.io.Serializable
import java.util.*

open class FoodItem(
        var itemName: String? = null,
        var itemPrice: String? = null,
        var itemCategory: String? = null,
        var itemRating: Long = 0,
) : Serializable {

    companion object {
        fun fromMap(dsp: MutableMap<String, Any>?, category: String?, itemName: String?): FoodItem? {
            val map = dsp as HashMap<String, Any>?
            return if (map != null) {
                var rating: Long = -1
                if (map.containsKey(FoodMenuDetails.RATING)) {
                    rating = map[FoodMenuDetails.RATING] as Long
                }
                FoodItem(
                        itemName, map[FoodMenuDetails.PRICE].toString(),
                        category,
                        rating,
                )
            } else {
                null
            }
        }
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as FoodItem

        if (itemName != other.itemName) return false
        if (itemPrice != other.itemPrice) return false
        if (itemCategory != other.itemCategory) return false
        if (itemRating != other.itemRating) return false

        return true
    }

    override fun hashCode(): Int {
        var result = itemName?.hashCode() ?: 0
        result = 31 * result + (itemPrice?.hashCode() ?: 0)
        result = 31 * result + (itemCategory?.hashCode() ?: 0)
        result = 31 * result + itemRating.hashCode()
        return result
    }
}